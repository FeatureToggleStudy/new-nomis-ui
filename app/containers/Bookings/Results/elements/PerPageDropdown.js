import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import LabelText from '@govuk-react/label-text'
import { SelectInput } from '@govuk-react/select'
import { SPACING } from '@govuk-react/constants'

const DropdownContainer = styled.div`
  margin-left: ${SPACING.SCALE_3};

  span {
    font-weight: 700;
  }
`

const PerPageDropdown = ({ handleChange, totalResults, perPage }) => (
  <DropdownContainer>
    <label htmlFor="perPage">
      <LabelText>Results per page</LabelText>
    </label>
    <SelectInput id="perPage" onChange={event => handleChange(event.target.value)} value={perPage}>
      <option value="10">10</option>
      <option value="20">20</option>
      <option value="50">50</option>
      <option value="100">100</option>
      <option value={totalResults}>All</option>
    </SelectInput>
  </DropdownContainer>
)

PerPageDropdown.propTypes = {
  handleChange: PropTypes.func.isRequired,
  totalResults: PropTypes.number.isRequired,
  perPage: PropTypes.number.isRequired,
}

export default PerPageDropdown
