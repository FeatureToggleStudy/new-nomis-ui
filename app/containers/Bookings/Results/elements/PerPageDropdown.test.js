import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import PerPageDropdown from './PerPageDropdown'

describe('<PerPageDropdown />', () => {
  const props = {
    handleChange: jest.fn(),
    totalResults: 999,
    perPage: 20,
  }

  it('should match the default snapshot', () => {
    const tree = renderer.create(<PerPageDropdown {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should trigger the handleChange function with the total results value when selected', () => {
    const wrapper = shallow(<PerPageDropdown {...props} />)

    wrapper.find('#perPage').simulate('change', { target: { value: props.totalResults } })
    expect(props.handleChange).toBeCalledWith(props.totalResults)
  })
})
