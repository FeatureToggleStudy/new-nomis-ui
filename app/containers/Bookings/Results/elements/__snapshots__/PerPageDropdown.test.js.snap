// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PerPageDropdown /> should match the default snapshot 1`] = `
<div
  className="sc-EHOje filMHp"
>
  <label
    htmlFor="perPage"
  >
    <span
      className="sc-bdVaJa fqDsBO"
    >
      Results per page
    </span>
  </label>
  <select
    className="sc-ifAKCX kDzsBS"
    id="perPage"
    onChange={[Function]}
    value={20}
  >
    <option
      value="10"
    >
      10
    </option>
    <option
      value="20"
    >
      20
    </option>
    <option
      value="50"
    >
      50
    </option>
    <option
      value="100"
    >
      100
    </option>
    <option
      value={999}
    >
      All
    </option>
  </select>
</div>
`;
